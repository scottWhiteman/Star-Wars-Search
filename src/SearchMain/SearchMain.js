import React from 'react';
import {Route, Link} from 'react-router-dom';
import { usePromiseTracker } from 'react-promise-tracker';
import Loader from 'react-loader-spinner';
import SearchBox from '../SearchBox/SearchBox';
import ResultsList from '../ResultsList/ResultsList';
import './SearchMain.css';

export default class SearchMain extends React.Component {
    state ={
        results: [],
        category: ''
    }
    
    Loading = () => {
        const { promiseInProgress } = usePromiseTracker();
        return (
            promiseInProgress &&
            <div className="Loading-Container">
                <h1>Searching </h1>
                <Loader type="ThreeDots" color="#eacc22" height="100" width="100" />
            </div>
        );
    }

    setData = (data, category) => {
        this.setState(
            {
                results : data,
                category
            }
        );
        console.log(this.state);
    }

    renderResultsList() {
        return (
            <>
                <Route
                    path="/"
                    render={routeProps => (
                        <ResultsList
                            results={this.state.results}
                            category={this.state.category}
                            {...routeProps}/>
                    )}
                />
            </>
        );
    }

    render() {
        return (
            <main className="Search-Main">
                <SearchBox setData={this.setData}/>
                <this.Loading />
                {this.state.results.length > 0 && this.renderResultsList()}
            </main>
        );
    }
}