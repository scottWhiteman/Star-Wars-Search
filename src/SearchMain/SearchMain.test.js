import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SearchMain from '../SearchMain/SearchMain';

describe("SearchMain Component", () => {
    it('renders SearchMain without crash', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SearchMain />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders to the UI', () => {
        const tree = renderer.create(<SearchMain />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})