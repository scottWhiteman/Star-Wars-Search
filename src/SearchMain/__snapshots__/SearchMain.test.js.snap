// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMain Component renders to the UI 1`] = `
<main
  className="Search-Main"
>
  <div
    className="Search-Container"
  >
    <form
      className="Search-Form"
      onSubmit={[Function]}
    >
      <input
        name="search"
        type="text"
      />
      <button
        type="submit"
      >
        Search!
      </button>
      <select
        name="category"
      >
        <option
          value="people"
        >
          Characters
        </option>
        <option
          value="species"
        >
          Species
        </option>
        <option
          value="films"
        >
          Films
        </option>
        <option
          value="planets"
        >
          Planets
        </option>
        <option
          value="starships"
        >
          Spaceships
        </option>
        <option
          value="vehicles"
        >
          Vehicles
        </option>
      </select>
    </form>
  </div>
</main>
`;
